src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1244 @@
+
+            var swarm = new SwarmService {LocalPeer = self};
+            var swarmConnections = 0;
+            swarm.ConnectionEstablished += (s, e) => { ++swarmConnections; };
+            await swarm.StartAsync();
+            try
+            {
+                var remotePeer = await swarm.ConnectAsync(peerBAddress);
+                Assert.AreEqual(1, swarmConnections);
+
+                // wait for swarmB to settle
+                var endTime = DateTime.Now.AddSeconds(3);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                        Assert.Fail("swarmB did not raise event.");
+                    if (swarmBConnections == 1)
+                        break;
+                    await Task.Delay(100);
+                }
+            }
+            finally
+            {
+                await swarm.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public void Connect_No_Transport()
+        {
+            var remoteId = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb";
+            MultiAddress remoteAddress = $"/ip4/127.0.0.1/ipfs/{remoteId}";
+            var swarm = new SwarmService {LocalPeer = self};
+            swarm.StartAsync().Wait();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() =>
+                {
+                    var _ = swarm.ConnectAsync(remoteAddress).Result;
+                });
+            }
+            finally
+            {
+                swarm.StopAsync().Wait();
+            }
+        }
+
+        [TestMethod]
+        public void Connect_Refused()
+        {
+            var remoteId = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb";
+            MultiAddress remoteAddress = $"/ip4/127.0.0.1/tcp/4040/ipfs/{remoteId}";
+            var swarm = new SwarmService {LocalPeer = self};
+            swarm.StartAsync().Wait();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() =>
+                {
+                    var _ = swarm.ConnectAsync(remoteAddress).Result;
+                });
+            }
+            finally
+            {
+                swarm.StopAsync().Wait();
+            }
+        }
+
+        [TestMethod]
+        public void Connect_Failure_Event()
+        {
+            var remoteId = "QmXFX2P5ammdmXQgfqGkfswtEVFsZUJ5KeHRXQYCTdiTAb";
+            MultiAddress remoteAddress = $"/ip4/127.0.0.1/tcp/4040/ipfs/{remoteId}";
+            var swarm = new SwarmService {LocalPeer = self};
+            Peer unreachable = null;
+            swarm.PeerNotReachable += (s, e) => { unreachable = e; };
+            swarm.StartAsync().Wait();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() =>
+                {
+                    var _ = swarm.ConnectAsync(remoteAddress).Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)